project/src/main/ui/chat/chat-library.gd

```diff
@@ -51,6 +51,27 @@ func load_chat_events_for_creature(creature: Creature, level_num: int = -1) -> C
 	return chat_tree
 
 
+"""
+Returns a ChatIcon enum representing the creature's next conversation.
+"""
+func chat_icon_for_creature(creature: Creature) -> int:
+	var result := ChatIcon.NONE
+	if creature == ChattableManager.instructor or creature == ChattableManager.player:
```

Comment:
Add some comments throughout this function, it's got a lot of branches which you might forget about.

Reply:
Added comments.